From bded311d32daa2339055341a7f1c1782ff39d047 Mon Sep 17 00:00:00 2001
From: Fabian Deutsch <fabian.deutsch@gmx.de>
Date: Sun, 1 Jan 2012 21:41:04 +0100
Subject: [PATCH 2/2] Add compiler option for ENABLE_USER_CODEMEM.

This option disbales non-user-dependent path checking at compile time. If enabled, only paths corresponding to a user are checked.

Signed-off-by: Fabian Deutsch <fabian.deutsch@gmx.de>
---
 configure.ac     |    4 ++++
 orc/Makefile.am  |    3 +++
 orc/orccodemem.c |    2 ++
 3 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 00e1916..4daee88 100644
--- a/configure.ac
+++ b/configure.ac
@@ -176,6 +176,10 @@ AM_CONDITIONAL(ENABLE_BACKEND_ARM, test "x$ENABLE_BACKEND_ARM" = "xyes")
 AM_CONDITIONAL(ENABLE_BACKEND_C64X, test "x$ENABLE_BACKEND_C64X" = "xyes")
 AM_CONDITIONAL(ENABLE_BACKEND_MIPS, test "x$ENABLE_BACKEND_MIPS" = "xyes")
 
+AC_ARG_ENABLE(user-codemem,
+  AC_HELP_STRING([--enable-user-codemem],[Force codemem allocation to be user dependent (default is no)]),
+    [], [enable_user_codemem=no])
+AM_CONDITIONAL(ENABLE_USER_CODEMEM, test "x$enable_user_codemem" = "xyes")
 
 AC_DEFINE(ORC_EXPORTS, 1, [Defined for compiling internal code])
 
diff --git a/orc/Makefile.am b/orc/Makefile.am
index 26263e0..887d36b 100644
--- a/orc/Makefile.am
+++ b/orc/Makefile.am
@@ -9,6 +9,9 @@ liborc_@ORC_MAJORMINOR@_la_LDFLAGS = \
 	-no-undefined -export-symbols-regex 'orc_'
 liborc_@ORC_MAJORMINOR@_la_CFLAGS = $(ORC_CFLAGS) \
 	-DORC_ENABLE_UNSTABLE_API
+if ENABLE_USER_CODEMEM
+liborc_@ORC_MAJORMINOR@_la_CFLAGS += -DORC_FORCE_USER_CODEMEM
+endif
 
 liborc_@ORC_MAJORMINOR@_la_SOURCES = \
 	orc.c \
diff --git a/orc/orccodemem.c b/orc/orccodemem.c
index 295a880..4a91e3e 100644
--- a/orc/orccodemem.c
+++ b/orc/orccodemem.c
@@ -280,12 +280,14 @@ orc_code_region_allocate_codemem (OrcCodeRegion *region)
 {
   const char *tmpdir;
 
+#ifndef ORC_FORCE_USER_CODEMEM
   tmpdir = getenv ("TMPDIR");
   if (tmpdir && orc_code_region_allocate_codemem_dual_map (region,
         tmpdir, FALSE)) return;
 
   if (orc_code_region_allocate_codemem_dual_map (region,
         "/tmp", FALSE)) return;
+#endif
 
   tmpdir = getenv ("XDG_RUNTIME_DIR");
   if (tmpdir && orc_code_region_allocate_codemem_dual_map (region,
-- 
1.7.7.6

